### PR TITLE
Improve paas-billing database connection handling

### PR DESCRIPTION
**Pull request by corlettb**
What
----

We have seen a long standing issue with lockups of the paas-billing components. These lockup point towards stale database connections. This change introduces a number of changes to make the connection more reliable.

- Upgrade of the postgres pq library
- Adding new configuration options DB_CONN_MAX_IDLE_TIME, DB_CONN_MAX_LIFETIME, DB_MAX_IDLE_CONNS to ensure db connections are cycled regularly.
- Add http status probe to collector.
- Updating http status checks to ping database.
- Stick to one db pool rather than having two.
 

How to review
-----

Successful run of acceptance tests can be viewed [here](https://deployer.dev04.dev.cloudpipeline.digital/teams/main/pipelines/create-cloudfoundry/jobs/deploy-paas-billing/builds/24).

Who can review
-----

One of the old timers familiar with the paas-billing code.

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

### CODE REVIEW

File: apiserver/server.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ type Config struct {
 	Logger lager.Logger
 	// EnablePanic will cause the server to crash on panic if set to true
 	EnablePanic bool
+	// Flag to indicate if only the health check should be exposed
+	StatusOnly bool
```
Comment (risicle) [inline apiserver/server.go:26]: I guess this name makes as much sense as any in context of everything else calling this `status`.

Comment (corlettb) [reply apiserver/server.go:26]: Want me to rename status here and the function call to healthz? I'd be tempted to change the url from / to healthz as well.

Comment (corlettb) [reply apiserver/server.go:26]: Okay can do. Let me make the update.

Comment (corlettb) [reply apiserver/server.go]: Done

**Review by risicle — APPROVED** [review]
*I* think this makes sense. Would be good to get @seb-szy to cast just one eye over it first though.

**Comment by risicle** [comment]
(sorry @seb-szy pinged you accidentally again)

**Review by schmie — APPROVED** [review]
LGTM 👍 Tests are happy in GA and `dev04`

**Review by risicle — APPROVED** [review]
I think this is safe to merge before https://github.com/alphagov/paas-cf/pull/3027/files

**Comment by risicle** [comment]
Ok maybe it failed but it failed safely.

**Pull request closed without merging**